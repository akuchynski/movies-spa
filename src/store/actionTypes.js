export default {

    GET_MOVIES_REQUEST: 'GET_MOVIES_REQUEST',
    GET_MOVIES_SUCCESS: 'GET_MOVIES_SUCCESS',
    GET_MOVIES_ERROR: 'GET_MOVIES_ERROR',

    GET_MOVIE_REQUEST: 'GET_MOVIE_REQUEST',
    GET_MOVIE_SUCCESS: 'GET_MOVIE_SUCCESS',
    GET_MOVIE_ERROR: 'GET_MOVIE_ERROR',

    CLOSE_MOVIE_DETAILS: 'CLOSE_MOVIE_DETAILS',
    SET_ACTIVE_FILTER: 'SET_ACTIVE_FILTER',

    ADD_MOVIE_SUCCESS: 'ADD_MOVIE_SUCCESS',
    ADD_MOVIE_ERROR: 'ADD_MOVIE_ERROR',

    UPDATE_MOVIE_SUCCESS: 'UPDATE_MOVIE_SUCCESS',
    UPDATE_MOVIE_ERROR: 'UPDATE_MOVIE_ERROR',

    DELETE_MOVIE_SUCCESS: 'DELETE_MOVIE_SUCCESS',
    DELETE_MOVIE_ERROR: 'DELETE_MOVIE_ERROR',

};