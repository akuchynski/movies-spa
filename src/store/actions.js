import actionTypes from "./actionTypes";

const getMoviesRequest = () => ({
    type: actionTypes.GET_MOVIES_REQUEST,
});

const getMoviesSuccess = (movies) => ({
    type: actionTypes.GET_MOVIES_SUCCESS,
    payload: movies
});

const getMoviesError = (errorMessage) => ({
    type: actionTypes.GET_MOVIES_ERROR,
    payload: errorMessage
});

const getMovieRequest = () => ({
    type: actionTypes.GET_MOVIE_REQUEST,
});

const getMovieSuccess = (movie) => ({
    type: actionTypes.GET_MOVIE_SUCCESS,
    payload: movie
});

const getMovieError = (errorMessage) => ({
    type: actionTypes.GET_MOVIE_ERROR,
    payload: errorMessage
});

const closeMovieDetails = () => ({
    type: actionTypes.CLOSE_MOVIE_DETAILS,
});

const setActiveFilter = (value) => ({
    type: actionTypes.SET_ACTIVE_FILTER,
    payload: value
});

export default {
    getMoviesRequest,
    getMoviesSuccess,
    getMoviesError,
    getMovieRequest,
    getMovieSuccess,
    getMovieError,
    closeMovieDetails,
    setActiveFilter,
};