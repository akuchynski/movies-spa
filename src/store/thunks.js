import axios from "axios";
import actions from "./actions";

export const getMovies = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByGenre = (genre) => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies', genre !== 'ALL' && { params: { searchBy: 'genres', filter: genre } })
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByDate = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies?sortBy=release_date&sortOrder=desc')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByRating = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies?sortBy=vote_average&sortOrder=desc')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMovieById = (id) => (dispatch) => {
    dispatch(actions.getMovieRequest());
    axios.get('http://localhost:4000/movies/' + id)
        .then((response) => dispatch(actions.getMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.getMovieError(error.message)));
};

export const addMovie = (movie) => (dispatch) => {
    movie.genres = movie.genres.map(item => item.value);
    axios.post('http://localhost:4000/movies', movie)
        .then((response) => dispatch(actions.addMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.addMovieError(error.message)));
};

export const updateMovie = (movieId, movie) => (dispatch) => {
    movie.id = movieId;
    movie.genres = movie.genres.map(item => item.value);
    axios.put('http://localhost:4000/movies', movie)
        .then((response) => dispatch(actions.updateMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.updateMovieError(error.message)));
};

export const deleteMovie = (id) => (dispatch) => {
    axios.delete('http://localhost:4000/movies/' + id)
        .then(() => dispatch(actions.deleteMovieSuccess()))
        .catch((error) => dispatch(actions.deleteMovieError(error.message)));
};