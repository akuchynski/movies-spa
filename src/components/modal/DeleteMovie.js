import React from "react";
import { useClickOutside } from "../../hooks/useClickOutside";
import { useDispatch, batch } from "react-redux";
import { deleteMovie, getMovies } from "../../store/thunks";
import actions from "../../store/actions";

export const DeleteMovieModal = ({ handleClose, movieId }) => {

    const ref = useClickOutside(handleClose);

    const dispatch = useDispatch();

    const handleDelete = () => {
        batch(() => {
            dispatch(deleteMovie(movieId));
            dispatch(actions.closeMovieDetails);
            dispatch(getMovies());
        });
        handleClose();
    };

    return (
        <div className="modal">
            <div className="delete-movie-modal" ref={ref}>
                <span className="close" onClick={handleClose}>&#x2715;</span>
                <h2>DELETE MOVIE</h2>
                <div className="delete-text">Are you sure you want to delete this movie?</div>
                <form>
                    <div className="modal-button">
                        <button className="submit-button" onClick={handleDelete}>CONFIRM</button>
                    </div>
                </form>
            </div>
        </div>
    );
};