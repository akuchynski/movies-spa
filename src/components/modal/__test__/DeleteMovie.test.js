import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import { MemoryRouter } from "react-router-dom";
import { store } from "../../../store/configureStore";
import { DeleteMovieModal } from "../DeleteMovie";

describe('movie delete modal component tests', () => {
    it('should render movie delete component', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <DeleteMovieModal />
                </MemoryRouter>
            </Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});