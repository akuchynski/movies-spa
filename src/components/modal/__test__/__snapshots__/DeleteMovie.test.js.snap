// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie delete modal component tests should render movie modal component 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="delete-movie-modal"
    >
      <span
        class="close"
      >
        âœ•
      </span>
      <h2>
        DELETE MOVIE
      </h2>
      <div
        class="delete-text"
      >
        Are you sure you want to delete this movie?
      </div>
      <form>
        <div
          class="modal-button"
        >
          <button
            class="submit-button"
          >
            CONFIRM
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
